### PR TITLE
LPS-159065 Allow edit External Reference Code value of an ObjectEntry through API

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-159065

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -2205,6 +2222,22 @@ private void _validateAccountEntry(
 				" does not belong to it"));
 	}
 
+	private void _validateExternalReferenceCode(
+			String externalReferenceCode, long companyId,
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2229]: Hi Paulo, could you sort the parameters please?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 2]
```diff
@@ -2004,6 +2020,114 @@ private void _testScope(long groupId, String scope, boolean expectSuccess)
...
+		catch (ObjectEntryValuesException.MustNotBeDuplicate
+					objectEntryValuesException) {
+
+			Assert.assertEquals(
+				"Duplicate value " +
+					String.valueOf(objectEntry1.getObjectEntryId()),
+				objectEntryValuesException.getMessage());
+		}
+
+		int notStringValue = RandomTestUtil.randomInt();
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:2111]: I was thinking that we could rename this variable. How about naming this var with what it means, like "randomInt" than calling it with the opposite meaning "notStringValue"? What do you think?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi @p-albuquerque, 
Nice work :smile: I just asked for simple adjustments!
Talk to me if you have any questions.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:2229]: okay

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java (hunk 2 above)
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:2111]: I agree

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:forward

**Pull request closed without merging**